system/lib/wasmfs/wasmfs.cpp

```diff
@@ -113,10 +113,76 @@ __wasi_errno_t __wasi_fd_read(
     const uint8_t* buf = iovs[i].buf;
     __wasi_size_t len = iovs[i].buf_len;
 
-    file.read(buf, len);
+    static_cast<DataFile::Handle&>(file).read(buf, len);
     num += len;
   }
   *nread = num;
   return __WASI_ERRNO_INVAL;
 }
+
+__wasi_fd_t __syscall_open(long pathname, long flags, long mode) {
+  // Get the corresponding OpenFile from the open file table
+  FileTable::Handle fileTable = FileTable::get();
```

Comment:
how about `auto` for things like this? I think it keeps the code shorter and more readable.